components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/MobileNumberVerificationHandler.java

```diff
@@ -95,11 +103,51 @@ public void handleEvent(Event event) throws IdentityEventException {
             }
             /* We need to empty 'MOBILE_NUMBER_PENDING_VALUE_CLAIM' because having a value in that claim implies
             a verification is pending. But verification is not enabled anymore. */
-            if (claims.containsKey(IdentityRecoveryConstants.MOBILE_NUMBER_CLAIM)) {
+            if (claims.containsKey(IdentityRecoveryConstants.MOBILE_NUMBER_CLAIM) ||
+                    claims.containsKey(IdentityRecoveryConstants.VERIFIED_MOBILE_NUMBERS_CLAIM)) {
                 invalidatePendingMobileVerification(user, userStoreManager, claims);
             }
             claims.remove(IdentityRecoveryConstants.VERIFY_MOBILE_CLAIM);
+            claims.remove(IdentityRecoveryConstants.VERIFIED_MOBILE_NUMBERS_CLAIM);
+
+            if (supportMultipleMobileNumbers) {
+                List<String> allMobileNumbers = Utils.getExistingClaimValue(userStoreManager, user,
+                        IdentityRecoveryConstants.MOBILE_NUMBERS_CLAIM);
+                if (claims.containsKey(IdentityRecoveryConstants.MOBILE_NUMBER_CLAIM) &&
+                        !allMobileNumbers.contains(claims.get(IdentityRecoveryConstants.MOBILE_NUMBER_CLAIM))) {
+                    throw new IdentityEventClientException(IdentityRecoveryConstants.ErrorMessages.
+                            ERROR_CODE_MOBILE_NUMBER_SHOULD_BE_INCLUDED_IN_MOBILE_NUMBERS_LIST.getCode(),
+                            IdentityRecoveryConstants.ErrorMessages
+                                    .ERROR_CODE_MOBILE_NUMBER_SHOULD_BE_INCLUDED_IN_MOBILE_NUMBERS_LIST.getMessage());
+                }
+            } else {
+                // Multiple mobile numbers per user support is disabled.
+                if (log.isDebugEnabled()) {
```

Comment:
We can remove the log.isDebugEnabled() check. log.debug() internally check the level. If there are no string concats, method calls within debug() part, we may not need debug log check.